### PR TITLE
update prettier, add tailwind formattin, add precommit hook and formal all

### PR DESCRIPTION
**Pull request by TomRadford**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,18 +50,21 @@
...
+    "lint-staged": "^15.2.0",
     "local-cors-proxy": "^1.1.0",
     "lodash.get": "^4.4.2",
     "lodash.orderby": "^4.6.0",
     "netlify": "^13.1.10",
     "postcss": "^8.4.12",
     "postcss-load-config": "^3.1.4",
-    "prettier": "^2.8.0",
-    "prettier-plugin-svelte": "^2.10.1",
+    "prettier": "^3.1.0",
```
Comment (TomRadford) [inline package.json:65]: prettier was a major version out of date 🫣
... v3 tested and still working great

File: package.json [hunk 2]
```diff
@@ -50,18 +50,21 @@
     "@typescript-eslint/parser": "^6.0.0",
     "autoprefixer": "^10.4.16",
     "eslint": "^8.28.0",
-    "eslint-config-prettier": "^8.5.0",
+    "eslint-config-prettier": "^9.1.0",
     "eslint-plugin-import": "^2.26.0",
     "eslint-plugin-storybook": "^0.6.15",
     "eslint-plugin-svelte": "^2.33.2",
+    "husky": "^8.0.3",
```
Comment (TomRadford) [inline package.json:57]: this is the tool for git hooks 🐶

File: package.json [hunk 3]
```diff
@@ -8,14 +8,14 @@
...
     "build": "vite build",
     "preview": "vite preview",
-    "prepare": "svelte-kit sync",
+    "prepare": "svelte-kit sync && husky install",
     "storybook": "storybook dev -p 6006 --no-open",
     "build-storybook": "storybook build",
     "chromatic": "NODE_OPTIONS='--max-old-space-size=4096' npx chromatic",
     "check": "svelte-kit sync && svelte-check --threshold warning --tsconfig ./tsconfig.json",
     "check:watch": "svelte-kit sync && svelte-check --tsconfig ./tsconfig.json --watch",
-    "lint": "prettier --plugin-search-dir . --check . && eslint .",
```
Comment (TomRadford) [inline package.json:17]: plugin-search-dir not needed in v3 since we describe plugins in the .prettierrx

File: package.json [hunk 4]
```diff
@@ -49,19 +49,22 @@
     "@typescript-eslint/eslint-plugin": "^6.0.0",
     "@typescript-eslint/parser": "^6.0.0",
     "autoprefixer": "^10.4.16",
-    "eslint": "^8.28.0",
-    "eslint-config-prettier": "^8.5.0",
+    "eslint": "^8.55.0",
```
Comment (TomRadford) [inline package.json:52]: updated eslint here while im at it

File: .husky/.gitignore [hunk 5]
```diff
@@ -0,0 +1 @@
+_
```
Comment (TomRadford) [inline .husky/.gitignore:1]: this is to ignore the `_` folder that husky generates with a shell script needed for commit hooks

**Review by matthampton-stock2shop — APPROVED** [review]
Looks good to me

**Review by matthampton-stock2shop — APPROVED** [review]
No review! Too many files.

**Pull request merged**